patches/0013-p256-armv8-optimizations

```diff
@@ -593,6 +595,19 @@ Index: aws-lc/third_party/boringssl/crypto/fipsmodule/ec/p256.c
 +  }
 +}
 +
++static int ecp_nistz256_scalar_to_montgomery_inv_vartime(const EC_GROUP *group,
++                                                 EC_SCALAR *out,
++                                                 const EC_SCALAR *in) {
++  assert(group->order.width == P256_LIMBS);
```

Comment:
should this be a hard check and not just a debug assert? Related to question above.

Reply:
This comments and other ones are on code that was copied from `p256-x86_64.h`. In this branch https://github.com/nebeid/aws-lc/tree/CA591-clean, the top commit is to remove the copied code and make the x86_64 code shared with AArch64. There will be only one change that ifdef out a variable that doesn't apply to AArch64 (ARMv8) (you asked about it in the test, I'll highlight it below).
So I won't change this code here and we can open stories to improve it.

Reply:
I see, it is existing code. Fair :)